time-ago.pipe.ts

```diff
@@ -6,9 +6,9 @@ import {Pipe, PipeTransform, NgZone, ChangeDetectorRef, OnDestroy} from "@angula
 export class TimeAgoPipe implements PipeTransform, OnDestroy {
 	private timer: number;
 	constructor(private changeDetectorRef: ChangeDetectorRef, private ngZone: NgZone) {}
-	transform(value:string) {
+	transform(value:string|number) {
 		this.removeTimer();
-		let d = new Date(value);
+		let d = typeof value === 'string' ? new Date(value) : new Date(value);
```

Comment:
The condition here is required unfortunately (at least in TypeScript 2.7.2) - this ensures the type guards treat the first Date(value) as string and the second as number. Without it, TypeScript compiler complains (it won't auto select the matching constructor unfortunately)